src/mem/cache/prefetch/stride.cc

```diff
@@ -174,9 +181,9 @@ Stride::calculatePrefetch(const PrefetchInfo &pfi,
             return;
         }
 
-        // Round strides up to atleast 1 cacheline
-        int prefetch_stride = new_stride;
-        if (abs(new_stride) < blkSize) {
+        // Round strides up to at least 1 cacheline
+        int prefetch_stride = entry->stride;
+        if (abs(prefetch_stride) < blkSize) {
             prefetch_stride = (new_stride < 0) ? -blkSize : blkSize;
```

Comment:
I think new_stride should have been replaced by prefetch_stride here too

Reply:
Yes, you are correct. Thanks for catching this error. I've fixed this